cluster-autoscaler/cloudprovider/aws/aws_cloud_provider_test.go

```diff
@@ -276,3 +306,15 @@ func TestBuildAsg(t *testing.T) {
 	assert.Equal(t, 222, asg.MaxSize())
 	assert.Equal(t, "test-name", asg.Name)
 }
+
+func TestAutoDiscoverASG(t *testing.T) {
+	provider := testProvider(t, testAwsManager)
+	node := "aws:///us-east-1a/test-instance-i"
+	nl := []*string{&node}
+
+	err := AutoDiscoverNodeGroup(provider, nl)
```

Comment:
Can we add a test case where we run `AutoDiscoverNodeGroup` with the mock autoscaling API returning 2 ASGs on the second time?